### DISCUSSION TITLE
SLEAP 1.3.3 identitity switches:  no cull to target option when tracking

### DISCUSSION THREAD

**Discussion by fmassaNU** [talmolab/sleap#1664] [Help!]
I'm using SLEAP 1.3.3 and I'm running into a problem with identity switches. Initially, I just had 20+ different track identities and this was too burdensome to correct. Someone suggested using the "cull to target" option, but either it is no longer available or I can't find it. Instead I simply set the maximum tracks to 2. On the surface this appears fine, but I'm having a problem changing track identities when there are switches. When I look at the tracks under the "delete tracks" option, it appears that there are many (15+) tracks that simply have the same name. This makes me unable to switch track identities using the crtl + (track #) option, because selecting the presumed correct track name often generates a completely random track rather than the one I intended. 
Is there an option when running tracking that can actually limit to 2 track identities? Is there anything I can do to select the correct "track_0" out of a sea of tracks all named "track_0"?

**Comment by Marielamend** [comment]
Hello @fmassaNU!

You are correct, the cull-to-target option does not appear in the 1.3.3 version.

1. Are you using a supervised ID model?
Background on supervised ID models: (provided by @roomrys)
>Supervised ID models require annotating not only the location of body parts, but also require the user to annotate the Track associated with an Instance. Then, the tracker is actually part of the model prediction. The usual (non-supervised ID) models just require annotating the body part locations and then the tracker is run as an additional step AFTER model prediction. The options that the user mentions are only for the added-on tracker step used in the usual (non-supervised ID) model.
 
 If so.. 
 Make sure you are following the [recommended annotation pipeline](https://sleap.ai/tutorials/initial-labeling.html) (which suggests that users predict on Labeling Suggestions).

There is a similar issue you can take a look at #1652 

Other suggestions:
-  > Delete Multiple Tracks... > Unused to see if that slims down the list.
![Screenshot 2024-01-22 124259](https://github.com/talmolab/sleap/assets/120222245/29afc831-eac1-4824-8d29-df242beef9cb)
-  You can also select “Connect Single Track Breaks” under “Post-tracker data cleaning” for your inference pipeline.


Hope this helps,
Mariela

**Reply by fmassaNU** [reply]
Hi Mariela, 
Thank you for the suggestions and information. 
I'm not sure if I'm using the supervised ID model or not. I am using the multi-animal top down model. 
I tried deleting unused tracks but didn't see any improvement. I also previously tried using the "connect single track breaks" option to no avail. 
Something I did notice though is that when I create a new file with only one video, the appropriate number of tracks are generated. It appears that new tracks are created for each video and that they have the same name. This leads me to believe that there is a disconnect in multiple video files where the controls are reflective of all tracks within a project rather than the video that is currently selected. I can see how this might be a useful feature for some situations, but it is problematic for my project. Unless there is a way around this, the only solution I can think of is to create new individual SLEAP files for every one of my videos and tracking on them separately.

**Reply by roomrys** [reply]
Hi @fmassaNU,

At least for now, it will be less of a headache proofreading if you have a separate project for each of the videos you are tracking across. Or if you predicted on a per video basis, proofreading to congregate all instance onto their desired tracks before running inference on the next video and repeating. Which sounds like a more complex procedure than anyone should ideally have to follow.

I can confirm, SLEAP generates new tracks for each video that inference is run on. This is because the tracker (for non supervised ID models) has conventionally required sequential frames to make any (somewhat-accurate) judgement of which `Track` an animal belongs to. SLEAP has no way of knowing how different videos relate to each other temporally hence why new `Track`s are used.

### Thoughts on the topic
The above reflects that the scope of a `Track` was designed to be within a `Video` - is this outdated? What if we allow reusing `Track`s across the entire `Labels` project changing the scope of a `Track`. Is there anything obviously wrong with this (unexpected changes in behavior)?

### Idea 1: Only display `Track`s in selected `Video`
I really like your idea of only displaying `Track`s which are in the current `Video` (in the hotkey-toggled legend, but perhaps also in the Tracks > Set Instance Track menu?).

### Idea 2: For the love of everyone, enforce uniqueness of `Track.name`s (for new files)
I [very strongly dislike](https://www.google.com/search?q=hate) that there is no restriction on uniqueness names for tracks. I would be a strong proponent of storing and saving to the slp file a set of track names which would help ensure that each `Track.name` is unique. This might introduce breaking changes to existing projects if we implement it in such a way that it is mandatory no matter what, but if we update our `FORMAT_ID` and also flexibly use this `track_names: Set[str]` based on the `FORMAT_ID`, I think we could get by without any breaking changes. Maybe we even give users the option to allow SLEAP to auto-append to existing duplicate track names to convert them to unique track names.

Liezl
